docs/design/persistent-volume-provisioning.md

```diff
@@ -0,0 +1,129 @@
+}
+
+// PersistentVolumeControllerSpec is the specification of a PV controller.
+// The controller creates volumes similar to the Template until the available count equals the replica count
+type PersistentVolumeControllerSpec struct {
+	// Replicas is the minimum number of available persistent volumes desired in the system
+	Replicas int `json:"replicas"`
+
+	// Selector is a label query over persistent volumes that should match the Replicas count.
+	Selector map[string]string `json:"selector"`
+
+	// Template is the object that describes the PV that will be created if
+	// insufficient replicas are detected.
+	Template *PersistentVolumeTemplateSpec `json:"template,omitempty"`
+}
+
+// PersistentVolumeControllerStatus represents the current status of a PVController
+type PersistentVolumeControllerStatus struct {
+	// BoundReplicas is the number of replicas of this volume that are currently bound to PersistentVolumeClaims
+	BoundReplicas int `json:"replicas"`
+	AvailableReplicas int `json:"replicas"`
+}
+
+// PersistentVolumeTemplateSpec describes the persistent volume to be created from a template
+type PersistentVolumeTemplateSpec struct {
+	// Metadata of the PVs created from this template.
+	ObjectMeta `json:"metadata,omitempty"`
+
+	// Spec defines a persistent volume.
+	Spec PersistentVolumeSpec `json:"spec,omitempty"`
+}
+
+```
+
+##### PersistentVolumeController API
+
+| Action | HTTP Verb | Path | Description |
+| ---- | ---- | ---- | ---- |
+| CREATE | POST | /api/{version}/persistentvolumecontrollers/ | Create instance of PersistentVolumeController|
+| GET | GET | /api/{version}persistentvolumecontrollers/{name} | Get instance of PersistentVolumeController |
+| UPDATE | PUT | /api/{version}/persistentvolumecontrollers/{name} | Update instance of PersistentVolumeController |
+| DELETE | DELETE | /api/{version}/persistentvolumecontrollers/{name} | Delete instance of PersistentVolumeController |
+| LIST | GET | /api/{version}/persistentvolumecontrollers | List instances of PersistentVolumeController|
+| WATCH | GET | /api/{version}/watch/persistentvolumecontrollers | Watch for changes to a PersistentVolumeController|
+
+
+#### PersistentVolumeControllerManager
+
+Use DeltaFIFO to watch PersistentVolumes of a particular type as defined by the controller's label selector.
+
+Add/Sync events inspect for binding (pv.Spec.ClaimRef) and tallies available/bound.
+
+Uupdate events inspect for binding and phase.  Any volume unbound from its claim will be available for recycling.
+
+```
+if controller.replicas > 0:
+    while availableCount < controller.replicas:
+        make(controller.spec.template)
+
+    while availableCount > controller.replicas:
+        recycle(v) // where v is choosen at random
+        
+for v in listOfVolumesToRecycle: //pv.Status.Phase = api.VolumeReleased
+        recycle(v)
+```
+
+NFS is a good example where controller.replicas will equal 0.  Dynamic creation will not take place, but the controller
```

Comment:
I think all analogies to ReplicationController are valid.    Manually provisioned PVs would ideally fall under the management of a PersistentVolumeController as a "replica" (I don't necessarily want to steal this name, unless it is particularly meaningful in this context.  You tell me).